### PR TITLE
Hosted AMP pages - styling

### PR DESCRIPTION
**Pull request by lps88**
## What does this change?
- Move the styles out of the html file and into scss file.
- Fix the styling for onward journeys for video and article pages.
- Fix the About popup to look like the web version.
- rationalise side-margins so it looks consistent

## What is the value of this and can you measure success?
- use variables, mixins, linting, minification etc for css
- bring OJ & popup styles up to date with web versions

## Does this affect other platforms - Amp, Apps, etc?
Just AMP. See https://github.com/guardian/frontend/pull/15154

## Screenshots

Hosted Video OJ:

image

Hosted Article OJ:

image

About popup (old/new):
image

## Request for comment
@NataliaLKB 
@guardian/labs-beta

### CODE REVIEW

File: static/src/stylesheets/amp/_hosted.scss [hunk 1]
```diff
@@ -0,0 +1,853 @@
...
+    top: 0;
+    z-index: 0;
+    background: linear-gradient(transparent 100px, rgba(0, 0, 0, .5));
+}
+
+.hosted__cta-wrapper {
+    padding-top: 140px;
+    position: relative;
+    z-index: 1;
+    margin-left: 20px;
```
Comment (Calanthe) [inline static/src/stylesheets/amp/_hosted.scss]: Since we have sass now, you can use variables defined in the https://github.com/guardian/frontend/blob/d4422b4537165424e70a898d150db4e806ba04d6/static/src/stylesheets/pasteup/grid-system/_src.scss file.

File: static/src/stylesheets/amp/_hosted.scss [hunk 2]
```diff
@@ -0,0 +1,853 @@
...
+}
+
+.hosted__cta-link:active,
+.hosted__next-page--tile:active {
+    text-decoration: none;
+}
+
+.hosted__text {
+    line-height: 1.2;
+    color: #767676;
```
Comment (Calanthe) [inline static/src/stylesheets/amp/_hosted.scss]: We have colours as variables

File: static/src/stylesheets/amp/_hosted.scss [hunk 3]
```diff
@@ -0,0 +1,853 @@
...
+}
+
+.hosted__cta-btn-text .inline-icon {
+    fill: #000000;
+    position: absolute;
+    top: 6px;
+    right: 16px;
+}
+
+.hosted__cta-btn-text .inline-icon svg {
```
Comment (Calanthe) [inline static/src/stylesheets/amp/_hosted.scss]: The nesting could be done in the sass way.

File: static/src/stylesheets/amp/_hosted.scss [hunk 4]
```diff
@@ -0,0 +1,853 @@
...
+    border-color: #ffffff;
+}
+
+.hosted-page--bright .survey-text__header .inline-cross {
+    fill: #ffffff;
+}
+
+.hosted__glogo {
+    display: inline-block;
+    padding: 4px 10px;
```
Comment (NataliaLKB) [inline static/src/stylesheets/amp/_hosted.scss]: Can we use our variables? Maybe `$gs-baseline/3 $gs-gutter`

File: static/src/stylesheets/amp/_hosted.scss [hunk 5]
```diff
@@ -0,0 +1,853 @@
...
+    background-color: #005689;
```
Comment (NataliaLKB) [inline static/src/stylesheets/amp/_hosted.scss]: Can we use our colour variables here?

File: static/src/stylesheets/amp/_hosted.scss [hunk 6]
```diff
@@ -0,0 +1,853 @@
...
+}
+
+.hosted__banner {
+    height: 200px;
+    position: relative;
+    background-size: cover;
+    background-position: left;
+}
+
+.hosted__banner:after {
```
Comment (NataliaLKB) [inline static/src/stylesheets/amp/_hosted.scss]: Would we be able to use `&:after` with the block above? I find scss much easier to read. Also much easier to create DRY code.

File: static/src/stylesheets/amp/_hosted.scss [hunk 7]
```diff
@@ -0,0 +1,853 @@
...
+.hosted__cta.button {
+    padding: 4px;
+    height: 26px;
+    width: 26px;
+    margin: 6px;
+    font-size: 14px;
+    line-height: 26px;
+}
+
+.hosted__cta.button svg {
```
Comment (NataliaLKB) [inline static/src/stylesheets/amp/_hosted.scss]: I believe the svg should now have a class like `hosted__cta__svg` which you can use instead of nesting

**Review by NataliaLKB — CHANGES_REQUESTED** [review]
Once this css is using our variables and written like scss this will be amazing!

File: static/src/stylesheets/amp/_hosted.scss [hunk 8]
```diff
@@ -0,0 +1,853 @@
...
+    .content__article-body p.intro + p::first-letter {
+        display: inline-block;
+        float: left;
+        color: #000000;
+        padding: 12px 16px 4px;
+        font-size: 480%;
+        line-height: 4.25rem;
+        width: 4.8125rem;
+        height: 4.8125rem;
+        margin-right: .5rem;
```
Comment (NataliaLKB) [inline static/src/stylesheets/amp/_hosted.scss]: Probably best to use `px` for code consistency. It will be converted to `rem` values anyway

**Comment by lps88** [comment]
@Calanthe @NataliaLKB I am going to keep working on the sass itself, just wanted to get the PR open to check that it's being imported in the right place etc?

**Comment by NataliaLKB** [comment]
@lps88 yes that looks good 😄 🚀

**Comment by lps88** [comment]
@NataliaLKB is there a way to conditionally include the hosted css? I seem to be hitting the css size limit... something like we do here?
https://github.com/guardian/frontend/blob/master/common/app/views/fragments/amp/stylesheets/main.scala.html#L261

**Comment by NataliaLKB** [comment]
@lps88 yes! The same way we conditionally load other css files. I can show you in person tomorrow morning if you prefer. Otherwise you can take a look at what I did in https://github.com/guardian/frontend/pull/15154

File: common/app/views/fragments/amp/stylesheets/fonts.scala.html [hunk 9]
```diff
@@ -27,8 +26,6 @@
 .guardian-egyptian-loaded .fc-container__header__description,
 .guardian-egyptian-loaded .paid-content--advertisement-feature,
 .guardian-egyptian-loaded .witness__button,
-.guardian-egyptian-loaded .hosted-article-page,
```
Comment (NataliaLKB) [inline common/app/views/fragments/amp/stylesheets/fonts.scala.html]: Do we still want to use the fonts for the hosted pages?

File: static/src/stylesheets/amp/_hosted.scss [hunk 10]
```diff
@@ -11,7 +11,7 @@ $header-height: 48px;
     text-align: right;
 }
 
-.hosted__glogo svg {
+.hosted__glogo .inline-logo__svg {
```
Comment (NataliaLKB) [inline static/src/stylesheets/amp/_hosted.scss]: I still think not nesting would be the most optimal for this. If you add any extra class to this svg, it will automatically add that class with `__svg` after it. But definitely better! 😄

File: common/app/views/fragments/amp/stylesheets/fonts.scala.html (hunk 9 above)
Comment (lps88) [inline common/app/views/fragments/amp/stylesheets/fonts.scala.html]: We used guardian egyptian at first, since we didn't have a guardian sans, but since it's commercial content we decided a sans serif font is better.

I haven't had time to look into this AMP font loading mechanism, probably will go into a different PR, this one already has some scope creep!

File: static/src/stylesheets/amp/_hosted.scss (hunk 10 above)
Comment (lps88) [inline static/src/stylesheets/amp/_hosted.scss]: Yeah, we're not passing very specific class names through to the svg so playing it safe here. Though I suppose when we start only loading this css in hosted pages it will be fine

**Comment by lps88** [comment]
Thanks for all the feedback @NataliaLKB, I'll come find you tomorrow to go over a few things if that's okay? Or come find me at 3-40 😄

File: static/src/stylesheets/amp/_hosted.scss [hunk 11]
```diff
@@ -0,0 +1,845 @@
...
+    .hosted__next-page-header--border {
+        margin-left: -$gs-gutter;
+        width: 150%;
+    }
+
+}
+
+.hosted-gallery-page__main-body {
+    position: absolute;
+    max-width: 800px;
```
Comment (Calanthe) [inline static/src/stylesheets/amp/_hosted.scss]: Use `gs-container` mixin maybe?

File: static/src/stylesheets/amp/_hosted.scss [hunk 12]
```diff
@@ -0,0 +1,845 @@
...
+    box-sizing: border-box;
+}
+
+.hosted-next-page__wrapper {
+    background: #ffffff;
+}
+
+.hosted__next-page-thumb {
+    height: 54px;
+    width: 90px;
```
Comment (Calanthe) [inline static/src/stylesheets/amp/_hosted.scss]: 90px = `$gs-column-width * 1.5`

**Comment by Calanthe** [comment]
👍

File: static/src/stylesheets/amp/_hosted.scss [hunk 13]
```diff
@@ -0,0 +1,848 @@
+$header-height: 48px;
```
Comment (NataliaLKB) [inline static/src/stylesheets/amp/_hosted.scss]: `$gs-row-height + $gs-baseline` or  `$gs-baseline * 4`

File: static/src/stylesheets/amp/_hosted.scss [hunk 14]
```diff
@@ -0,0 +1,848 @@
...
+.hosted__glogo {
+    display: inline-block;
+    padding: $gs-baseline/3 $gs-gutter/2;
+    background-color: $guardian-brand;
+    text-align: right;
+}
+
+.hosted__guardian-logo__svg {
+    fill: #ffffff;
+    width: 115px;
```
Comment (NataliaLKB) [inline static/src/stylesheets/amp/_hosted.scss]: There are still quite a lot of magic numbers in this PR. Do they fit into our grid system at all?

File: static/src/stylesheets/amp/_hosted.scss [hunk 15]
```diff
@@ -0,0 +1,848 @@
...
+    margin: 0;
+    text-align: left;
+    color: rgba(255, 255, 255, .5);
+}
+
+.hostedbadge {
+    position: absolute;
+}
+
+.hostedbadge__logo.hostedbadge__logo {
```
Comment (NataliaLKB) [inline static/src/stylesheets/amp/_hosted.scss]: Just checking, is this intenstional? If a element has the same classname twice, do this thing?

File: static/src/stylesheets/amp/_hosted.scss [hunk 16]
```diff
@@ -0,0 +1,848 @@
...
+    margin-left: $gs-gutter;
+}
+
+.hosted__cta-btn-wrapper {
+    padding-top: 90px;
+}
+
+.hosted__cta.button,
+.hosted__cta.button:focus {
+    color: #333333;
```
Comment (NataliaLKB) [inline static/src/stylesheets/amp/_hosted.scss]: This is `neutral-1` http://guardian.github.io/pasteup-palette/demo/

File: static/src/stylesheets/amp/_hosted.scss [hunk 17]
```diff
@@ -0,0 +1,848 @@
...
+
+.hosted__header {
+    @include f-textSans;
+    width: 100%;
+    height: $header-height;
+    background-color: #333333;
+    position: relative;
+}
+
+.has-no-flex .hosted__header .hostedbadge {
```
Comment (NataliaLKB) [inline static/src/stylesheets/amp/_hosted.scss]: Just a question because, what does "has no flex" mean?

File: static/src/stylesheets/amp/_hosted.scss [hunk 18]
```diff
@@ -0,0 +1,848 @@
...
+            z-index: 2;
+            background: linear-gradient(to top, rgba(51, 51, 51, .4) 0%, rgba(51, 51, 51, .4) 2%, transparent 100%);
+        }
+    }
+
+    &.hosted-page--bright .content__article-body p.intro + p::first-letter {
+        color: #ffffff;
+    }
+
+    .gs-container {
```
Comment (NataliaLKB) [inline static/src/stylesheets/amp/_hosted.scss]: I think we normally advise against adding things to `gs-container`, is this needed and or can we put this on something else?

File: static/src/stylesheets/amp/_hosted.scss [hunk 19]
```diff
@@ -0,0 +1,848 @@
...
+
+    .caption {
+        position: absolute;
+        left: 10px;
+        top: 390px;
+        right: unset;
+        padding: 0;
+    }
+
+    figure {
```
Comment (NataliaLKB) [inline static/src/stylesheets/amp/_hosted.scss]: This also seems quite generic. Is this something that is needed, and why?

File: static/src/stylesheets/amp/_hosted.scss [hunk 20]
```diff
@@ -0,0 +1,848 @@
...
+    }
+
+    .hosted__next-page-header--border {
+        margin-left: -$gs-gutter;
+        width: 150%;
+    }
+
+}
+
+.hosted-gallery-page__main-body {
```
Comment (NataliaLKB) [inline static/src/stylesheets/amp/_hosted.scss]: Would it make sense to have hosted galleries have its own stylesheet? This one is massive and seems like it is touching a lot of different types of things. Might be worth breaking it down more.

You could event conditionally only load it on things that are hosted and galleries.

On that note though, galleries aren't currently supported in AMP, how are you doing them and what do they look like?

**Comment by NataliaLKB** [comment]
This is getting much better! Still missing some places where we should be using vars. 

Also, I don't know why there is so many magic numbers. As we have a grid system, normally magic numbers aren't needed. In places they are, I usually tell people to put a comment with a reason why they need them. That would be really long for you to do in this PR though. 

Instead feel free to just explain in this PR why magic numbers are needed, and feel free to let me know when you have updated this PR.

Really great job cleaning things up!

File: static/src/stylesheets/amp/_hosted.scss (hunk 20 above)
Comment (lps88) [inline static/src/stylesheets/amp/_hosted.scss]: It uses something called `amp-carousel`, not a perfect imitation of our main gallery page but you can see it here: https://amp.theguardian.com/advertiser-content/chester-zoo-act-for-wildlife/what-we-fight-for

File: static/src/stylesheets/amp/_hosted.scss (hunk 14 above)
Comment (lps88) [inline static/src/stylesheets/amp/_hosted.scss]: this is the logo:

image

The svg has width/height attributes which make it too big for the space. I'm not sure how to shrink it other than explicitly sizing it like this!

File: static/src/stylesheets/amp/_hosted.scss (hunk 20 above)
Comment (lps88) [inline static/src/stylesheets/amp/_hosted.scss]: It was built really recently, it's completely separate from other galleries really, so made sense to do it alongside our other hosted pages. Not sure how much it will get used! There's still some missing features but we don't have a lot of live galleries to work with so leaving it as is for now.

Comment (lps88) [inline static/src/stylesheets/amp/_hosted.scss]: But yes, it might make sense to have separate gallery stylesheet, I'll look into it.

**Comment by lps88** [comment]
@NataliaLKB Are there any more changes you think are important for this PR? I think I've got as much grid system goodness into it as is useful 🔳 📏 🔲

File: static/src/stylesheets/amp/_hosted-article.scss [hunk 21]
```diff
@@ -0,0 +1,120 @@
...
+
+    .meta__extras {
+        border-top: none;
+    }
+
+    .social__item--pinterest {
+        display: inline-block;
+    }
+
+    .content__article-body p {
```
Comment (NataliaLKB) [inline static/src/stylesheets/amp/_hosted-article.scss]: You can make lines 67 - 89 more Sassy. like this:

```
.content__article-body {
    p {
       ...
         &.intro ....
    }
    ...
}
```

File: static/src/stylesheets/amp/_hosted.scss [hunk 22]
```diff
@@ -0,0 +1,474 @@
...
+        content: '';
+        width: 100%;
+        height: 100%;
+        position: absolute;
+        left: 0;
+        top: 0;
+        z-index: 0;
+        background: linear-gradient(transparent 100px, rgba(0, 0, 0, .5));
+    }
+    .hosted__cta-wrapper {
```
Comment (NataliaLKB) [inline static/src/stylesheets/amp/_hosted.scss]: Does this have to be nested? Can this be combined with line the block in 68?

File: static/src/stylesheets/amp/_hosted.scss [hunk 23]
```diff
@@ -0,0 +1,474 @@
...
+    background: transparent;
+}
+
+.hosted__header {
+    $text-colour: #e3e3e3;
+
+    .survey-container {
+        position: absolute;
+        top: 20px;
+        left: 20px;
```
Comment (NataliaLKB) [inline static/src/stylesheets/amp/_hosted.scss]: `$gs-gutter`

File: static/src/stylesheets/amp/_hosted.scss [hunk 24]
```diff
@@ -0,0 +1,474 @@
...
+        right: 20px;
```
Comment (NataliaLKB) [inline static/src/stylesheets/amp/_hosted.scss]: `$gs-gutter`

File: static/src/stylesheets/amp/_hosted.scss [hunk 25]
```diff
@@ -0,0 +1,474 @@
...
+        padding-top: 16px;
+        background-color: $text-colour;
+        line-height: 10px;
+    }
+
+    .survey-text__header {
+        font-size: 24px;
+        font-weight: bold;
+        margin: 0;
+        padding: 6px 0 0 10px;
```
Comment (NataliaLKB) [inline static/src/stylesheets/amp/_hosted.scss]: top badding: `$gs-baseline / 2`

**Review by NataliaLKB — APPROVED** [review]
Please fix my last few comments before merging. But this is much better and much more readable! Thanks for addressing all my other concerns.
